scripts/esbuild/dev.js

```diff
@@ -1,15 +1,25 @@
 /* eslint-disable @typescript-eslint/no-var-requires */
 const esbuild = require('esbuild');
 const commonConfig = require('./common-config');
+const { publishPackage } = require('yalc');
+const { join } = require('path');
+
+const publishWithYalc = () => publishPackage({ workingDir: join(process.cwd() || ''), push: true });
 
 esbuild
   .build({
     ...commonConfig,
     watch: {
       onRebuild(error, result) {
         if (error) console.error('Watch build failed:', error);
-        else console.log('Watch build succeeded:', result);
+        else {
+          console.log('Watch build succeeded:', result);
+          publishWithYalc();
```

Comment:
Maybe we could log the publish step has finished after line 17

Reply:
btw, what if the call to  `publishWithYalc` fails? That would be a very bad DevX... maybe we should at least wrap the call in a catch clause, wdyt?

Reply:
Considering I have the package linked in 2 projects (uhc-portal and stolostron/console), the output of a build is following:
```
Build succeeded and started watching { errors: [], warnings: [], stop: [Function: stop] }
openshift-assisted-ui-lib@2.4.0 published in store.
Pushing openshift-assisted-ui-lib@2.4.0 in /root/assisted-ui
Package openshift-assisted-ui-lib@2.4.0 linked ==> /root/assisted-ui/node_modules/openshift-assisted-ui-lib
Pushing openshift-assisted-ui-lib@2.4.0 in /root/open-cluster-management-console/frontend
Package openshift-assisted-ui-lib@2.4.0 linked ==> /root/open-cluster-management-console/frontend/node_modules/openshift-assisted-ui-lib
```

Isn't that sufficient?